website/server/controllers/api-v3/hall.js

```diff
@@ -177,8 +177,8 @@ api.updateHero = {
       _.set(hero, updateData.itemPath, updateData.itemVal); // Sanitization at 5c30944 (deemed unnecessary)
     }
 
-    if (updateData.auth && _.isBoolean(updateData.auth.blocked)) {
-      hero.auth.blocked = updateData.auth.blocked;
+    if (updateData.auth && _.isBoolean(updateData.auth.blocked) && updateData.auth.blocked === true) {
```

Comment:
It looks like this line could be:

```js
if (updateData.auth && updateData.auth.blocked === true) {
// ...
}
```

All the cases where `_.isBoolean(updateData.auth.blocked)` is `false`, will also be caught by `updateData.auth.blocked === true`.